src/code_wars/next_bigger_number.py

```diff
@@ -0,0 +1,38 @@
+# pylint: disable=invalid-name
+import math
+
+
+def find_i_of_n_with_larger_m_to_right(numbers):
+    n_with_larger_to_right = None
+    for i in range(len(numbers) - 1, 0, -1):
+        left, right = numbers[i - 1], numbers[i]
```

Comment:
I am so lost with this right here! When I put this into ipython:

numbers = [123]
left, right=numbers, numbers[0]

after I hit enter I don't get an error and when I check type 
left - I get [123]
right - I get [123]

but I don't get why left is [123]
what is python doing if I just type 
x, y=[foo], foo[0]?????

Reply:
The debug can't seem to play nice with what is happening so I can't step into the function and watch it that way easily. But I tried to manually do the for loop in ipython with numbers = [1, 2, 3, 4]

`for i in range(len(numbers) - 1, 0, -1):
   left, right = numbers[i-1], numbers[i]
   print(f"left: {left} right: {numbers[i-1]} numbers[{i}]: {numbers[i]}")`

and the output I get is:
left: 3 right: 3 numbers[3]: 4
left: 2 right: 2 numbers[2]: 3
left: 1 right: 1 numbers[1]: 2

but I still can't figure out what is assigning left. It has to be something simple I am missing.

Reply:
```python
left, right = numbers[i - 1], numbers[i]
```
is almost the same as
```python
left = numbers[i - 1]
right = numbers[i]
```

It could be a little different under some circumstances. Consider:
```python
x = "abc"
y = "ghi"
x, y = y, x
print(x)
print(y)
```
Swaps just fine, but the same can't be written:
```python
x = "abc"
y = "ghi"
x = y 
y = x
print(x)
print(y)
```
you would have to put in a temp variable to store the value like:
 ```python
 x = "abc"
y = "ghi"
temp = x
x = y 
y = temp
print(x)
print(y)
```

Reply:
Think your print statement there should be `print(f"left: {left} right: {right} numbers[{i}]: {numbers[i]}")` instead of `right: {numbers[i-1]}`

Reply:
oh wow - that is slick. Just to clarify, essentially this is what it's doing

w, x, y, z = (w assignment), (x assignment), (y assignment), (z assignment)

Reply:
> Think your print statement there should be `print(f"left: {left} right: {right} numbers[{i}]: {numbers[i]}")` instead of `right: {numbers[i-1]}`

that assumption is based on the flawed belief that I had an idea of what I was doing - lol

I was intentionally doing the print statement that way because I was way off on what I thought was going on and I was trying to investigate but I think I got it now. I mean, I don't get it at all. I just think I am getting that one specific thing. I am still very lost as to what is actually happening as far as the actual solution.